RedisCluster/samples/V1/CloudRedisClusterClient/list_locations.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * GENERATED CODE WARNING
+ * This file was automatically generated - do not edit!
+ */
+
+require_once __DIR__ . '/../../../vendor/autoload.php';
+
+// [START redis_v1_generated_CloudRedisCluster_ListLocations_sync]
+use Google\ApiCore\ApiException;
+use Google\ApiCore\PagedListResponse;
+use Google\Cloud\Location\ListLocationsRequest;
+use Google\Cloud\Location\Location;
+use Google\Cloud\Redis\Cluster\V1\Client\CloudRedisClusterClient;
+
+/**
+ * Lists information about the supported locations for this service.
+ *
+ * This sample has been automatically generated and should be regarded as a code
+ * template only. It will require modifications to work:
+ *  - It may require correct/in-range values for request initialization.
+ *  - It may require specifying regional endpoints when creating the service client,
+ *    please see the apiEndpoint client configuration option for more details.
+ */
+function list_locations_sample(): void
+{
+    // Create a client.
+    $cloudRedisClusterClient = new CloudRedisClusterClient();
+
+    // Prepare the request message.
+    $request = new ListLocationsRequest();
+
+    // Call the API and handle any network failures.
+    try {
+        /** @var PagedListResponse $response */
+        $response = $cloudRedisClusterClient->listLocations($request);
+
+        /** @var Location $element */
+        foreach ($response as $element) {
+            printf('Element data: %s' . PHP_EOL, $element->serializeToJsonString());
+        }
+    } catch (ApiException $ex) {
+        printf('Call failed with message: %s' . PHP_EOL, $ex->getMessage());
+    }
+}
```

Comment:
Can you test this out with gRPC as well as rest? I noticed the documentation [only has REST](https://cloud.google.com/memorystore/docs/cluster/apis), which makes me concerned that it won't work for gRPC.

Reply:
Verified that it works for gRPC as well as REST.